CMakeLists.txt

```diff
@@ -66,6 +66,12 @@ else()
 endif()
 configure_file(src/preload/rr_page.ld.in src/preload/rr_page.ld @ONLY)
 
+include(CheckCCompilerFlag)
+CHECK_C_COMPILER_FLAG("-fmacro-prefix-map=$(srctree)/=" SUPPORTS_MACRO_PREFIX_MAP)
+if (SUPPORTS_MACRO_PREFIX_MAP)
+  set(FLAGS_COMMON "${FLAGS_COMMON} -fmacro-prefix-map=${CMAKE_SOURCE_DIR}/=")
```

Comment:
I think this should be added to the test flags instead and not the common flags. For the installed binary, the distribution package builder may set this to point to the actual installation location of the source files instead.

Reply:
I don't understand, sorry. The goal is to make __FILE__ more readable. I think it makes sense to do this in distribution packages as well as local builds. What's the problem?

Reply:
Distribution can set this to the actual installation path of the source (when the source code is installed as part of the debug package.

Reply:
What do you mean by "this"? `CMAKE_SOURCE_DIR`?

Reply:
fmacro-prefix-map

Reply:
In particular, the value it maps the cmake source path into. Here the mapped value is just empty.

Reply:
I don't think it's particularly useful for distributions to set this. It only appears in rr messages, it's not changing debuginfo as far as I know.

Reply:
Having said that, this change doesn't seem to work locally.

Reply:
OK, it only works with the `make` backend, not `ninja`...

Reply:
Fixed that with b9f4eea09d1038347fc009f8ce6c7752e0951c06